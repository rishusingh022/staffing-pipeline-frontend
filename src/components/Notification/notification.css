.notification-card{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    background-color: #fde7e7;
    font-size:"1rem";
    padding: "15px";
    width: 100%;
}
.close-button{
    color: #a41523;
    background-color:#fde7e7;
    padding: 10px;
    margin-right: 10px ;
    border-radius: 50%;
}
.close-button:hover{
    background-color: rgba(256, 256, 256, 30%);
    color: #a41523;
    cursor: pointer;
}
.notification-card .message{
    font-size: 1rem;
    font-weight: 400;
    font-family: 'Inter';
    color: #a41523;
    margin: 10px;
}