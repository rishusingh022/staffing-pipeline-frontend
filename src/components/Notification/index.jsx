import React from 'react';

import './notification.css';
import propTypes from 'prop-types';

export default function Notification({ message, handleClose, success }) {
  return (
    <div className={`notification-card ${success ? 'success-bg' : 'error-bg'}`}>
      <div className="message">
        <p className={`${success ? 'success-text' : 'error-text'}`}>{message}</p>
      </div>
      <div
        className={`close-button ${success ? 'success-bg success-text' : 'error-bg error-text'} `}
        data-testid="X"
        onClick={() => handleClose()}>
        X
      </div>
    </div>
  );
}
Notification.propTypes = {
  message: propTypes.string,
  handleClose: propTypes.func,
  success: propTypes.bool,
};
