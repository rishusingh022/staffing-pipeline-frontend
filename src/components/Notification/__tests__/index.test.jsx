import Notification from '..';
import { render, screen, fireEvent } from '@testing-library/react';
describe('Notification', () => {
  it('Notification snapshot', () => {
    const { asFragment } = render(<Notification message="test" handleClose={jest.fn} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders notification', () => {
    render(<Notification message="test" handleClose={jest.fn} />);
    expect(screen.getByText('test')).toBeTruthy();
  });
  it("should close when 'X' is clicked", () => {
    const mockHandleCloseFunc = jest.fn();
    render(<Notification message="test" handleClose={mockHandleCloseFunc} />);
    fireEvent.click(screen.getByTestId('X'));
    expect(mockHandleCloseFunc).toHaveBeenCalled();
  });
});
