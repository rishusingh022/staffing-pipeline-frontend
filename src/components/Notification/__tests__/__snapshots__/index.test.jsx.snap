// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Notification snapshot 1`] = `
<DocumentFragment>
  <div
    class="notification-card"
  >
    <div
      class="message"
    >
      <p>
        test
      </p>
    </div>
    <div
      class="close-button"
      data-testid="X"
    >
      X
    </div>
  </div>
</DocumentFragment>
`;
